frontend/apps/reader/modules/readersearch.lua

```diff
@@ -51,33 +60,99 @@ function ReaderSearch:onShowFulltextSearchInput()
+                        if self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText()) ~= 0 then
+                            UIManager:show(InfoMessage:new{ text = _("Error in regular expression!") })
+                        else
+                            UIManager:close(self.input_dialog)
+                            self:onShowSearchDialog(self.input_dialog:getInputText(), 1, self.use_regex, self.case_insensitive)
+                        end
                     end,
                 },
                 {
                     text = forward_text,
                     is_enter_default = true,
                     callback = function()
                         if self.input_dialog:getInputText() == "" then return end
-                        UIManager:close(self.input_dialog)
-                        self:onShowSearchDialog(self.input_dialog:getInputText(), 0)
+                        self.last_search_text = self.input_dialog:getInputText()
+                        self.use_regex = self.check_button_regex.checked
+                        self.case_insensitive = self.check_button_case.checked
+                        if self.use_regex and self.ui.document:checkRegex(self.input_dialog:getInputText()) ~= 0 then
+                            UIManager:show(InfoMessage:new{ text = _("Error in regular expression!") })
+                        else
+                            UIManager:close(self.input_dialog)
+                            self:onShowSearchDialog(self.input_dialog:getInputText(), 0, self.use_regex, self.case_insensitive)
+                        end
                     end,
                 },
             },
         },
     }
+
+   -- checkboxes
+    self.check_button_regex = self.check_button or CheckButton:new{
+        text = _("Regular expression"),
+        face = Font:getFace("smallinfofont"),
+        checked = self.use_regex,
+        callback = function()
+            if not self.check_button_regex.checked then
+                self.check_button_regex:check()
+            else
+                self.check_button_regex:unCheck()
+            end
+            self.input_dialog:onShow()
+        end,
+        padding = self.input_dialog.padding,
+        margin = self.input_dialog.margin,
+        bordersize = self.input_dialog.bordersize,
+    }
+    self.check_button_case = self.check_button or CheckButton:new{
+        text = _("Case insensitive"),
+        face = Font:getFace("smallinfofont"),
+        checked = self.case_insensitive,
+        callback = function()
+            if not self.check_button_case.checked then
+                self.check_button_case:check()
+            else
+                self.check_button_case:unCheck()
+            end
+            self.input_dialog:onShow()
+        end,
+        padding = self.input_dialog.padding,
+        margin = self.input_dialog.margin,
+        bordersize = self.input_dialog.bordersize,
```

Comment:
Looks like InputDialog doesn't have any of these properties, so there must be `nil`.
Looks ok though:
![image](https://user-images.githubusercontent.com/24273478/123510366-5593c080-d67b-11eb-91e5-a41a18afebdf.png)

I would somehow invert them (case should be more useful than regexp), and I would may be make them both unchecked by default - so `[ ] Case sensitive` with negated checks if we keep the original internal `self.case_insensitive`